app/main.py

```diff
@@ -1,3 +1,151 @@
 class Distance:
-    # Write your code here
-    pass
+    """
+    Клас для роботи з дистанцією в кілометрах.
+    """
+
+    def __init__(self, km: int) -> None:
+        """
+        Ініціалізує об'єкт Distance з кількістю кілометрів.
+
+        :param km: Дистанція в кілометрах
+        (ціле число або число з плаваючою точкою).
+
+        """
+        if not isinstance(km, (int, float)):
+            raise TypeError("Distance must be a number (int or float).")
+        if km < 0:
+            raise ValueError("Distance cannot be negative.")
+        self.km = km
+
+    def __str__(self) -> str:
+        """
+        Повертає текстове представлення об'єкта для користувача.
+
+        :return: Рядок виду 'Distance: {self.km} kilometers.'
+        """
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        """
+        Повертає технічне представлення об'єкта для розробників.
+
+        :return: Рядок виду 'Distance(km={self.km})'.
+        """
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int) -> None:
```

Comment:
The `other` parameter in the `__add__` method should be annotated to accept both `Distance` and numeric types (int, float).